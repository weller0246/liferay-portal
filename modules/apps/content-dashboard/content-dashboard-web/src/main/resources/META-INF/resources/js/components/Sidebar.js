/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayButtonWithIcon} from '@clayui/button';
import ClayLayout from '@clayui/layout';
import {useTimeout} from '@liferay/frontend-js-react-web';
import classNames from 'classnames';
import React, {useContext, useEffect, useState} from 'react';

const SidebarContext = React.createContext();

const noop = () => {};

const SidebarBody = ({children, className}) => {
	return (
		<div className={classNames('sidebar-body', className)}>{children}</div>
	);
};

const SidebarHeader = ({title}) => {
	const {onClose} = useContext(SidebarContext);

	return (
		<section className="sidebar-header">
			<ClayLayout.ContentRow className="sidebar-section">
				<ClayLayout.ContentCol expand>
					<h2 className="font-weight-bold mb-0 pr-2 small">
						{title}
					</h2>
				</ClayLayout.ContentCol>

				<ClayLayout.ContentCol>
					<ClayButtonWithIcon
						aria-label="Close"
						className="mt-n2 text-secondary"
						displayType="unstyled"
						onClick={onClose}
						symbol="times"
					/>
				</ClayLayout.ContentCol>
			</ClayLayout.ContentRow>
		</section>
	);
};

const Sidebar = ({children, onClose = noop, open = true}) => {
	const [isOpen, setIsOpen] = useState(false);

	const delay = useTimeout();

	// Wait until the component is rendered to show it so animation happens

	useEffect(() => {
		if (open !== false) {
			delay(() => setIsOpen(true), 100);
		}
		else {
			setIsOpen(false);
		}
	}, [delay, open]);

	useEffect(() => {
		if (isOpen) {
			document.body.classList.add('sidebar-open');
		}
		else {
			document.body.classList.remove('sidebar-open');
		}
	}, [isOpen]);

	return (
		<div className="content-dashboard sidebar sidebar-light sidebar-sm">
			<SidebarContext.Provider value={{onClose}}>
				{children}
			</SidebarContext.Provider>
		</div>
	);
};

Sidebar.Body = SidebarBody;
Sidebar.Header = SidebarHeader;

export {SidebarContext};
export default Sidebar;
