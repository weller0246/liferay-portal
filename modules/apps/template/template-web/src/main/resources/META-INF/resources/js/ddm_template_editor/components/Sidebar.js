/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayButtonWithIcon} from '@clayui/button';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, {useEffect} from 'react';

import {ElementsSidebarPanel} from './ElementsSidebarPanel';
import PropertiesSidebarPanel from './PropertiesSidebarPanel';
import {PANEL_IDS} from './panelIds';

export default function Sidebar({
	selectedSidebarPanelId,
	setSelectedSidebarPanelId,
}) {
	useEffect(() => {
		const sideNavigation = Liferay.SideNavigation.instance(
			document.querySelector('.product-menu-toggle')
		);

		if (sideNavigation) {
			if (selectedSidebarPanelId && sideNavigation.visible()) {
				sideNavigation.hide();
			}

			const sideNavigationListener = sideNavigation.on(
				'openStart.lexicon.sidenav',
				() => setSelectedSidebarPanelId(null)
			);

			return () => {
				sideNavigationListener.removeListener();
			};
		}
	}, [selectedSidebarPanelId, setSelectedSidebarPanelId]);

	return (
		<div className="ddm_template_editor__App-sidebar">
			<div
				className={classNames(
					'ddm_template_editor__App-sidebar-content',
					{
						open: selectedSidebarPanelId,
					}
				)}
			>
				<ElementsSidebarPanel
					className={classNames({
						'd-none': PANEL_IDS.elements !== selectedSidebarPanelId,
					})}
				/>

				<PropertiesSidebarPanel
					className={classNames({
						'd-none':
							PANEL_IDS.properties !== selectedSidebarPanelId,
					})}
				/>
			</div>
			<div className="ddm_template_editor__App-sidebar-buttons pt-3">
				<ClayButtonWithIcon
					aria-pressed={PANEL_IDS.elements === selectedSidebarPanelId}
					borderless
					className={classNames('mb-2', {
						active: PANEL_IDS.elements === selectedSidebarPanelId,
					})}
					data-tooltip-align="left"
					displayType="secondary"
					monospaced
					onClick={() => {
						setSelectedSidebarPanelId((selectedSidebarPanelId) =>
							selectedSidebarPanelId === PANEL_IDS.elements
								? null
								: PANEL_IDS.elements
						);
					}}
					outline
					symbol="list-ul"
					title={Liferay.Language.get('elements')}
				/>

				<ClayButtonWithIcon
					aria-pressed={
						PANEL_IDS.properties === selectedSidebarPanelId
					}
					borderless
					className={classNames('mb-2', {
						active: PANEL_IDS.properties === selectedSidebarPanelId,
					})}
					data-tooltip-align="left"
					displayType="secondary"
					monospaced
					onClick={() => {
						setSelectedSidebarPanelId((selectedSidebarPanelId) =>
							selectedSidebarPanelId === PANEL_IDS.properties
								? null
								: PANEL_IDS.properties
						);
					}}
					outline
					symbol="cog"
					title={Liferay.Language.get('properties')}
				/>
			</div>
		</div>
	);
}

Sidebar.propTypes = {
	selectedSidebarPanelId: PropTypes.string,
	setSelectedSidebarPanelId: PropTypes.func.isRequired,
};
