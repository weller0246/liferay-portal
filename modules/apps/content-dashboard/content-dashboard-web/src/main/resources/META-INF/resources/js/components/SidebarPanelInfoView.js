/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayIcon from '@clayui/icon';
import ClayLabel from '@clayui/label';
import ClaySticker from '@clayui/sticker';
import classnames from 'classnames';
import React from 'react';

import DocumentPreview from './DocumentPreview';
import ItemLanguages from './ItemLanguages';
import Sidebar from './Sidebar';

const formatDate = (date, languageTag) => {
	return (
		date &&
		languageTag &&
		Intl.DateTimeFormat(languageTag, {
			day: 'numeric',
			hour: 'numeric',
			hour12: true,
			minute: 'numeric',
			month: 'short',
			year: 'numeric',
		}).format(new Date(date))
	);
};

const SidebarPanelInfoView = ({
	categories = [],
	className,
	classPK,
	createDate,
	data = {},
	languageTag = 'en',
	modifiedDate,
	specificFields = {},
	subType,
	tags = [],
	title,
	user,
	versions = [],
	viewURLs = [],
}) => {
	const stickerColor = parseInt(user.userId, 10) % 10;

	const {
		description,
		downloadURL,
		extension,
		fileName,
		previewImageURL,
		previewURL,
		size,
		viewURL,
	} = specificFields;

	const itemDates = [
		{
			text: formatDate(data['display-date']?.value, languageTag),
			title: Liferay.Language.get('display-date'),
		},
		{
			text: formatDate(createDate, languageTag),
			title: Liferay.Language.get('creation-date'),
		},
		{
			text: formatDate(modifiedDate, languageTag),
			title: Liferay.Language.get('modified-date'),
		},
		{
			text: formatDate(data['expiration-date']?.value, languageTag),
			title: Liferay.Language.get('expiration-date'),
		},
		{
			text: formatDate(data['review-date']?.value, languageTag),
			title: Liferay.Language.get('review-date'),
		},
		{
			text: classPK,
			title: Liferay.Language.get('id'),
		},
	];

	const isADocument =
		className === 'com.liferay.portal.kernel.repository.model.FileEntry';

	const documentIsAFile =
		isADocument &&
		!!downloadURL &&
		!!extension &&
		parseInt(size?.split(' ')[0], 10) > 0;

	const documentUsesPreview = !!previewImageURL || documentIsAFile;

	const showTaxonomies = !!categories?.length || !!tags?.length;

	return (
		<>
			<Sidebar.Header title={title} />

			<Sidebar.Body>
				<div className="mb-2 sidebar-section">
					{documentIsAFile && (
						<p className="mb-1 text-secondary">{fileName}</p>
					)}

					<p className="mb-1 text-secondary">{subType}</p>

					{versions.map((version) => (
						<div key={version.version}>
							<ClayLabel displayType="info">
								{Liferay.Language.get('version')}{' '}
								{version.version}
							</ClayLabel>

							<ClayLabel displayType={version.statusStyle}>
								{version.statusLabel}
							</ClayLabel>
						</div>
					))}
				</div>

				<div className="sidebar-dl sidebar-section">
					<ClaySticker
						className={classnames('sticker-user-icon', {
							[`user-icon-color-${stickerColor}`]: !user.url,
						})}
						shape="circle"
					>
						{user.url ? (
							<img
								alt={`${user.name}.`}
								className="sticker-img"
								src={user.url}
							/>
						) : (
							<ClayIcon symbol="user" />
						)}
					</ClaySticker>
					<span className="ml-2 text-secondary">{user.name}</span>
				</div>

				{documentUsesPreview && (
					<DocumentPreview
						documentSrc={previewImageURL}
						documentTitle={title}
						downloadURL={downloadURL}
						isFile={documentIsAFile}
						viewURL={viewURL}
					/>
				)}

				{description && (
					<div className="sidebar-section">
						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('description')}
						</h5>
						<p className="text-secondary">{description}</p>
					</div>
				)}

				{showTaxonomies && (
					<h6 className="font-weight-semi-bold sidebar-section-subtitle text-uppercase">
						{Liferay.Language.get('categorization')}
					</h6>
				)}

				{!!categories.length && (
					<div className="c-mb-4 sidebar-dl sidebar-section">
						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('categories')}
						</h5>

						<p>
							{categories.map((category) => (
								<ClayLabel
									displayType="secondary"
									key={category}
									large
								>
									{category}
								</ClayLabel>
							))}
						</p>
					</div>
				)}

				{!!tags.length && (
					<div className="c-mb-4 sidebar-dl sidebar-section">
						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('tags')}
						</h5>

						<p>
							{tags.map((tag) => (
								<ClayLabel
									displayType="secondary"
									key={tag}
									large
								>
									{tag}
								</ClayLabel>
							))}
						</p>
					</div>
				)}

				<h6 className="font-weight-semi-bold sidebar-section-subtitle text-uppercase">
					{Liferay.Language.get('details')}
				</h6>

				{documentIsAFile && (
					<div className="sidebar-section">
						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('url')}
						</h5>
						<p className="text-secondary">{previewURL}</p>

						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('extension')}
						</h5>
						<p className="text-secondary">{extension}</p>

						<h5 className="font-weight-semi-bold mb-1">
							{Liferay.Language.get('size')}
						</h5>
						<p className="text-secondary">{size}</p>
					</div>
				)}

				{!!itemDates.length &&
					itemDates.map(
						({text, title}) =>
							text &&
							title && (
								<div
									className="c-mb-4 sidebar-dl sidebar-section"
									key={title}
								>
									<h5 className="font-weight-semi-bold mb-1">
										{title}
									</h5>

									<p className="text-secondary">{text}</p>
								</div>
							)
					)}

				{!!viewURLs.length && !isADocument && (
					<ItemLanguages urls={viewURLs} />
				)}
			</Sidebar.Body>
		</>
	);
};

export default SidebarPanelInfoView;
