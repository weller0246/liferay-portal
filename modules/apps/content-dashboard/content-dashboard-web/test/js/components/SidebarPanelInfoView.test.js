/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';

import Sidebar from '../../../src/main/resources/META-INF/resources/js/components/Sidebar';
import SidebarPanelInfoView from '../../../src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView';
import {
	mockedFileDocumentProps,
	mockedImageDocumentProps,
	mockedNoTaxonomies,
	mockedProps,
	mockedUser,
	mockedVideoShortcutDocumentProps,
} from '../mocks/props';

const _getSidebarComponent = (props) => {
	return (
		<Sidebar>
			<SidebarPanelInfoView {...props} />
		</Sidebar>
	);
};

describe('SidebarPanelInfoView', () => {
	afterEach(() => {
		jest.clearAllMocks();
		cleanup();
	});

	it('renders', () => {
		const {asFragment} = render(_getSidebarComponent(mockedProps));

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders sidebar panel with proper info for a basic web content', () => {
		const {container, getByText} = render(
			_getSidebarComponent(mockedProps)
		);

		expect(getByText('Basic Web Content Title')).toBeInTheDocument();
		expect(getByText('Basic Web Content')).toBeInTheDocument();
		expect(getByText('version 1.6')).toBeInTheDocument();
		expect(getByText('Approved')).toBeInTheDocument();
		expect(getByText('version 1.7')).toBeInTheDocument();
		expect(getByText('Draft')).toBeInTheDocument();

		expect(getByText('Kate Williams')).toBeInTheDocument();
		const avatar = container.querySelector('.lexicon-icon-user');
		expect(avatar).toBeTruthy();

		expect(getByText('languages-translated-into')).toBeInTheDocument();
		expect(getByText('en-US')).toBeInTheDocument();
		expect(getByText('default')).toBeInTheDocument();
		expect(getByText('es-ES')).toBeInTheDocument();
		expect(getByText('fr-FR')).toBeInTheDocument();
		expect(getByText('pt-BR')).toBeInTheDocument();

		expect(getByText('tags')).toBeInTheDocument();
		expect(getByText('tag1')).toBeInTheDocument();
		expect(getByText('tag2')).toBeInTheDocument();

		expect(getByText('categories')).toBeInTheDocument();
		expect(getByText('Crime')).toBeInTheDocument();
		expect(getByText('Romantic')).toBeInTheDocument();
		expect(getByText('Classics')).toBeInTheDocument();
		expect(getByText('Fantasy')).toBeInTheDocument();
		expect(getByText('Science Fiction')).toBeInTheDocument();
		expect(getByText('Advanced')).toBeInTheDocument();
		expect(getByText('Education')).toBeInTheDocument();
		expect(getByText('Decision')).toBeInTheDocument();

		expect(getByText('display-date')).toBeInTheDocument();
		expect(getByText('Jul 27, 2020, 10:53 AM')).toBeInTheDocument();

		expect(getByText('creation-date')).toBeInTheDocument();
		expect(getByText('Jul 27, 2020, 10:50 AM')).toBeInTheDocument();

		expect(getByText('modified-date')).toBeInTheDocument();
		expect(getByText('Jul 27, 2020, 10:50 AM')).toBeInTheDocument();

		expect(getByText('expiration-date')).toBeInTheDocument();
		expect(getByText('Jul 28, 2020, 10:00 AM')).toBeInTheDocument();

		expect(getByText('review-date')).toBeInTheDocument();
		expect(getByText('Jul 27, 2020, 2:14 PM')).toBeInTheDocument();

		expect(getByText('id')).toBeInTheDocument();
		expect(getByText('38070')).toBeInTheDocument();

		expect(getByText('categorization')).toBeInTheDocument();
		expect(getByText('details')).toBeInTheDocument();
	});

	it('renders sidebar panel with proper info for a basic web content', () => {
		const {queryByText} = render(
			_getSidebarComponent({...mockedProps, ...mockedNoTaxonomies})
		);

		expect(queryByText('categorization')).not.toBeInTheDocument();
	});

	it('renders sidebar panel with proper info for an image document', () => {
		const {container, getByText, queryByText} = render(
			_getSidebarComponent({
				...mockedProps,
				...mockedImageDocumentProps,
			})
		);
		const previewFigureTag = container.querySelector('figure');

		expect(
			previewFigureTag.classList.contains('document-preview-figure')
		).toBe(true);

		expect(getByText('Basic Document')).toBeInTheDocument();
		expect(getByText('Mocked description')).toBeInTheDocument();
		expect(getByText('download')).toBeInTheDocument();
		expect(getByText('size')).toBeInTheDocument();
		expect(getByText('url')).toBeInTheDocument();

		expect(
			queryByText('languages-translated-into')
		).not.toBeInTheDocument();
	});

	it('renders sidebar panel with proper info for a video shortcut document', () => {
		const {container, getByText, queryByText} = render(
			_getSidebarComponent({
				...mockedProps,
				...mockedVideoShortcutDocumentProps,
			})
		);
		const previewFigureTag = container.querySelector('figure');

		expect(
			previewFigureTag.classList.contains('document-preview-figure')
		).toBe(true);

		expect(getByText('Mocked description')).toBeInTheDocument();
		expect(getByText('External Video Shortcut')).toBeInTheDocument();

		expect(queryByText('download')).not.toBeInTheDocument();
		expect(queryByText('size')).not.toBeInTheDocument();
		expect(queryByText('filename')).not.toBeInTheDocument();
		expect(queryByText('url')).not.toBeInTheDocument();
		expect(
			queryByText('languages-translated-into')
		).not.toBeInTheDocument();
	});

	it('renders sidebar panel with proper info for a file', () => {
		const {container, getByText, queryByText} = render(
			_getSidebarComponent({
				...mockedProps,
				...mockedFileDocumentProps,
			})
		);
		const previewFigureTag = container.querySelector('figure');

		expect(previewFigureTag).toBe(null);

		expect(getByText('Basic Document')).toBeInTheDocument();
		expect(getByText('download')).toBeInTheDocument();

		expect(
			queryByText('languages-translated-into')
		).not.toBeInTheDocument();
	});

	it('renders sidebar panel with proper info if author has avatar', () => {
		const {container} = render(
			_getSidebarComponent({
				...mockedProps,
				...mockedUser,
			})
		);

		const avatar = container.querySelector('.lexicon-icon-user');
		expect(avatar).toBeFalsy();

		const image = container.querySelector('.sticker-img');
		expect(image).toBeTruthy();
	});
});
