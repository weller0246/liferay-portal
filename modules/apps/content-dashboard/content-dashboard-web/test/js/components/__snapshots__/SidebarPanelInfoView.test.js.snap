// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarPanelInfoView renders 1`] = `
<DocumentFragment>
  <div
    class="content-dashboard sidebar sidebar-light sidebar-sm"
  >
    <section
      class="sidebar-header"
    >
      <div
        class="sidebar-section autofit-row"
      >
        <div
          class="autofit-col autofit-col-expand"
        >
          <h2
            class="font-weight-bold mb-0 pr-2 small"
          >
            Basic Web Content Title
          </h2>
        </div>
        <div
          class="autofit-col"
        >
          <button
            aria-label="Close"
            class="mt-n2 text-secondary btn btn-monospaced btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-times"
              role="presentation"
            >
              <use
                xlink:href="#times"
              />
            </svg>
          </button>
        </div>
      </div>
    </section>
    <div
      class="sidebar-body"
    >
      <div
        class="mb-2 sidebar-section"
      >
        <p
          class="mb-1 text-secondary"
        >
          Basic Web Content
        </p>
        <div>
          <span
            class="label label-info"
          >
            <span
              class="label-item label-item-expand"
            >
              version 1.6
            </span>
          </span>
          <span
            class="label label-success"
          >
            <span
              class="label-item label-item-expand"
            >
              Approved
            </span>
          </span>
        </div>
        <div>
          <span
            class="label label-info"
          >
            <span
              class="label-item label-item-expand"
            >
              version 1.7
            </span>
          </span>
          <span
            class="label label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Draft
            </span>
          </span>
        </div>
      </div>
      <div
        class="sidebar-dl sidebar-section"
      >
        <span
          class="sticker sticker-user-icon user-icon-color-6 sticker-circle"
        >
          <span
            class="sticker-overlay"
          >
            <svg
              class="lexicon-icon lexicon-icon-user"
              role="presentation"
            >
              <use
                xlink:href="#user"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 text-secondary"
        >
          Kate Williams
        </span>
      </div>
      <h6
        class="font-weight-semi-bold sidebar-section-subtitle text-uppercase"
      >
        categorization
      </h6>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          categories
        </h5>
        <p>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Crime
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Romantic
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Classics
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Fantasy
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Science Fiction
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Advanced
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Education
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              Decision
            </span>
          </span>
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          tags
        </h5>
        <p>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              tag1
            </span>
          </span>
          <span
            class="label label-lg label-secondary"
          >
            <span
              class="label-item label-item-expand"
            >
              tag2
            </span>
          </span>
        </p>
      </div>
      <h6
        class="font-weight-semi-bold sidebar-section-subtitle text-uppercase"
      >
        details
      </h6>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          display-date
        </h5>
        <p
          class="text-secondary"
        >
          Jul 27, 2020, 10:53 AM
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          creation-date
        </h5>
        <p
          class="text-secondary"
        >
          Jul 27, 2020, 10:50 AM
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          modified-date
        </h5>
        <p
          class="text-secondary"
        >
          Jul 27, 2020, 10:56 AM
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          expiration-date
        </h5>
        <p
          class="text-secondary"
        >
          Jul 28, 2020, 10:00 AM
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          review-date
        </h5>
        <p
          class="text-secondary"
        >
          Jul 27, 2020, 2:14 PM
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold mb-1"
        >
          id
        </h5>
        <p
          class="text-secondary"
        >
          38070
        </p>
      </div>
      <div
        class="c-mb-4 sidebar-dl sidebar-section"
      >
        <h5
          class="font-weight-semi-bold"
        >
          languages-translated-into
        </h5>
        <div
          class="autofit-row autofit-row-center"
        >
          <div
            class="inline-item-before autofit-col"
          >
            <svg
              class="lexicon-icon lexicon-icon-en-us c-mt-1"
              role="presentation"
            >
              <use
                xlink:href="#en-us"
              />
            </svg>
          </div>
          <div
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-row autofit-row-center"
            >
              <div
                class="inline-item-before small autofit-col"
              >
                en-US
              </div>
              <div
                class="autofit-col"
              >
                <span
                  class="label d-inline label-info"
                >
                  <span
                    class="label-item label-item-expand"
                  >
                    default
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="autofit-col"
          >
            <a
              class="link-monospaced link-outline link-outline-borderless link-outline-secondary"
              data-tooltip-align="top"
              href="http://localhost:8080/en-US/web/guest/-/basic-web-content-title"
              title="view"
            >
              <svg
                class="lexicon-icon lexicon-icon-view"
                role="presentation"
              >
                <use
                  xlink:href="#view"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="autofit-row autofit-row-center"
        >
          <div
            class="inline-item-before autofit-col"
          >
            <svg
              class="lexicon-icon lexicon-icon-es-es c-mt-1"
              role="presentation"
            >
              <use
                xlink:href="#es-es"
              />
            </svg>
          </div>
          <div
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-row autofit-row-center"
            >
              <div
                class="inline-item-before small autofit-col"
              >
                es-ES
              </div>
              <div
                class="autofit-col"
              />
            </div>
          </div>
          <div
            class="autofit-col"
          >
            <a
              class="link-monospaced link-outline link-outline-borderless link-outline-secondary"
              data-tooltip-align="top"
              href="http://localhost:8080/es-ES/web/guest/-/basic-web-content-title"
              title="view"
            >
              <svg
                class="lexicon-icon lexicon-icon-view"
                role="presentation"
              >
                <use
                  xlink:href="#view"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="autofit-row autofit-row-center"
        >
          <div
            class="inline-item-before autofit-col"
          >
            <svg
              class="lexicon-icon lexicon-icon-fr-fr c-mt-1"
              role="presentation"
            >
              <use
                xlink:href="#fr-fr"
              />
            </svg>
          </div>
          <div
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-row autofit-row-center"
            >
              <div
                class="inline-item-before small autofit-col"
              >
                fr-FR
              </div>
              <div
                class="autofit-col"
              />
            </div>
          </div>
          <div
            class="autofit-col"
          >
            <a
              class="link-monospaced link-outline link-outline-borderless link-outline-secondary"
              data-tooltip-align="top"
              href="http://localhost:8080/fr-FR/web/guest/-/basic-web-content-title"
              title="view"
            >
              <svg
                class="lexicon-icon lexicon-icon-view"
                role="presentation"
              >
                <use
                  xlink:href="#view"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="autofit-row autofit-row-center"
        >
          <div
            class="inline-item-before autofit-col"
          >
            <svg
              class="lexicon-icon lexicon-icon-pt-br c-mt-1"
              role="presentation"
            >
              <use
                xlink:href="#pt-br"
              />
            </svg>
          </div>
          <div
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-row autofit-row-center"
            >
              <div
                class="inline-item-before small autofit-col"
              >
                pt-BR
              </div>
              <div
                class="autofit-col"
              />
            </div>
          </div>
          <div
            class="autofit-col"
          >
            <a
              class="link-monospaced link-outline link-outline-borderless link-outline-secondary"
              data-tooltip-align="top"
              href="http://localhost:8080/pt-BR/web/guest/-/basic-web-content-title"
              title="view"
            >
              <svg
                class="lexicon-icon lexicon-icon-view"
                role="presentation"
              >
                <use
                  xlink:href="#view"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
